src/document.rs

```diff
@@ -51,24 +51,22 @@ pub(crate) fn extract_value(any: &PyAny) -> PyResult<Value> {
     if let Ok(b) = any.extract::<Vec<u8>>() {
         return Ok(Value::Bytes(b));
     }
-    if let Ok(dict) = any.downcast::<PyDict>() {
-        if let Ok(json) = pythonize::depythonize(dict) {
-            return Ok(Value::Object(json));
-        }
+    if let Ok(json) = pythonize::depythonize_bound(any.clone()) {
+        return Ok(Value::Object(json));
     }
     Err(to_pyerr(format!("Value unsupported {any:?}")))
 }
 
 pub(crate) fn extract_value_for_type(
-    any: &PyAny,
+    any: Bound<PyAny>,
     tv_type: tv::schema::Type,
     field_name: &str,
 ) -> PyResult<Value> {
     // Helper function to create `PyErr`s returned by this function.
     fn to_pyerr_for_type<'a, E: std::error::Error>(
         type_name: &'a str,
         field_name: &'a str,
-        any: &'a PyAny,
+        any: Bound<'a, PyAny>,
```

Comment:
This might work using `&'a Bound<'_, PyAny>` as well and thereby avoid some of the short-lived clones? Note that the reference `'py` in the `Bound<'py, T>` is the lifetime for which the GIL is held, not necessarily the lifetime for which a Rust borrow exists. The old GIL-bound references tended to conflate these often distinct lifetimes.

Reply:
Yep, it worked. However I had to not include the inner lifetime parameter at all

```rust
        any: &'a Bound<PyAny>,
```